writers/writer.go

```diff
@@ -57,22 +66,79 @@ func (c *consumer) consume(m *nats.Msg) {
 	}
 	var msgs []senml.Message
 	for _, v := range norm {
-		if c.channelExists(v.Channel) {
+		if c.channelExists(v.Channel) && c.subtopicExists(v.Subtopic) {
 			msgs = append(msgs, v)
 		}
 	}
 
+	if msgs == nil {
+		return
+	}
+
```

Comment:
This check should be done when you start the service. You should subscribe to NATS subjects that aree in the config file (channel + subtopic). If channel or/and subject are empty `"*"` there is no filter to apply and you can subscribe to all subjects as you are doing now.

Reply:
Ok, I will see what I can did. I will probably modifying in first inside main's writer (cassandra, influx, ..) and after see how we can manage this point.